sql/catalyst/src/main/java/org/apache/spark/sql/catalyst/expressions/UnsafeRow.java

```diff
@@ -171,8 +122,64 @@ private void setNotNullAt(int i) {
   }
 
   @Override
-  public void update(int ordinal, Object value) {
-    throw new UnsupportedOperationException();
+  public void update(int i, Object value) {
```

Comment:
There seems to be an unstated assumption that this method will only be called to update String, Binary, or Object-valued columns and not primitive columns.  Can you add a comment stating this explicitly?